### PR TITLE
Feat/#300 dict diff filters

### PR DESCRIPTION
**Pull request by ciaranschutte**
- adds comparison filters (uses own `button` instead of `uikit/button` because overriding styles is proving rather obtuse)
- data hookup for filtering still to be implemented, will depends a lot on data needed for changes page, will talk to @kcullion tomorrow

### CODE REVIEW

File: website/src/pages/dictionary/index.js [hunk 1]
```diff
@@ -185,37 +196,63 @@ function DataDictionary() {
...
+                attributeBool = true;
+              }
+
+              const compareVal = get(field, 'compare', '');
+              const { additionIsActive, deletionIsActive, updateIsActive } = compareFilters;
+
+              const comparisonBool =
+                (additionIsActive && compareVal === 'addition') ||
+                (deletionIsActive && compareVal === 'deletion') ||
+                (updateIsActive && compareVal === 'update');
```
Comment (hlminh2000) [inline website/src/pages/dictionary/index.js:251]: there seems to be some scattered "language" that all seem to refer to the same things... `additionIsActive`, `addition` and `ADDITIONS` are all different manifestation of the same concept. Any chance we can reduce this variant? something like making `compareFilterTypes` something like `{ ADDITION: "ADDITION" ... }` maybe?

File: website/src/pages/dictionary/index.js [hunk 2]
```diff
@@ -185,37 +196,63 @@ function DataDictionary() {
...
+                (attribute === TAG_TYPES.conditional && Boolean(dependsOn)) ||
+                (attribute === TAG_TYPES.required && required) ||
+                attribute === '' ||
+                attribute === NO_ACTIVE_FILTER
+              ) {
+                attributeBool = true;
+              }
+
+              const compareVal = get(field, 'compare', '');
+              const { additionIsActive, deletionIsActive, updateIsActive } = compareFilters;
```
Comment (hlminh2000) [inline website/src/pages/dictionary/index.js:246]: `compareFilters` needs to be added to the `useMemo` dependency :)

File: website/src/pages/dictionary/index.js [hunk 3]
```diff
@@ -185,41 +196,67 @@ function DataDictionary() {
...
+             * Remove this after demo
+             * !!!!!!!!!!!!!!!!!!!!!!
+             */
+            .map((fields, i) => {
+              if (i === 2) {
+                return { ...fields, ...{ compare: 'addition' } };
+              } else if (i === 3) {
+                return { ...fields, ...{ compare: 'deletion' } };
+              } else if (i === 7) {
+                return { ...fields, ...{ compare: 'update' } };
```
Comment (hlminh2000) [inline website/src/pages/dictionary/index.js:211]: should we also just make this uppercase? maybe use `compareFilterTypes.ADDITION` for consistency

File: website/src/pages/dictionary/index.js [hunk 4]
```diff
@@ -185,41 +196,67 @@ function DataDictionary() {
...
+                attributeBool = true;
+              }
+
+              const compareVal = get(field, 'compare', '');
+              const { ADDITION, DELETION, ACTIVE } = compareFilters;
+
+              const comparisonBool =
+                (ADDITION && compareVal === 'addition') ||
+                (DELETION && compareVal === 'deletion') ||
+                (ACTIVE && compareVal === 'update');
```
Comment (hlminh2000) [inline website/src/pages/dictionary/index.js:251]: `compareFilterTypes.ADDITION`?

**Comment by ciaranschutte** [comment]
@hlminh2000 fixed language

did some cleanup too

the major part being the `dictionary` and `diff` state.

We really need to figure out how we're getting the diffs because a lot of the UI depends on having a "diff type" key at the top of the data structure, we can talk more for next sprint. 

Really really really need to add TS to this project :D

**Comment by ciaranschutte** [comment]
@hlminh2000 
- rebased to get latest docusaurus update to ensure it still builds/runs
- fixed build issue, component was in `pages` folder which is special folder for pages :(

Going ahead with merging this

**Pull request merged**